src/main/java/com/powsybl/openloadflow/network/impl/AbstractLfBus.java

```diff
@@ -415,22 +415,26 @@ public void addHvdc(LfHvdc hvdc) {
         hvdcs.add(Objects.requireNonNull(hvdc));
     }
 
-    private static double dispatchQ(List<LfGenerator> generatorsThatControlVoltage, boolean reactiveLimits, double qToDispatch) {
+    protected static double dispatchQ(List<LfGenerator> generatorsThatControlVoltage, boolean reactiveLimits, double qToDispatch) {
         double residueQ = 0;
-        double calculatedQ = qToDispatch / generatorsThatControlVoltage.size();
+        if (generatorsThatControlVoltage.isEmpty()) {
```

Comment:
This check is not needed because it is done just before calling this method.

Reply:
This was my suggestion and was to ensure that through debug log User is aeare that there has indeed been no redistribution

Reply:
Plus we should always consider

Separation of concerns based implementation